Duplicati/UnitTest/SymLinkTests.cs

```diff
@@ -0,0 +1,157 @@
+                    {
+                        systemIO.DirectoryDelete(dir, true);
+                    }
+                    foreach (var file in systemIO.EnumerateFiles(this.RESTOREFOLDER))
+                    {
+                        systemIO.FileDelete(file);
+                    }
+                    c.DeleteAllRemoteFiles();
+                    if (systemIO.FileExists(this.DBFILE))
+                    {
+                        systemIO.FileDelete(this.DBFILE);
+                    }
+                    if (systemIO.FileExists($"{this.DBFILE}-journal"))
+                    {
+                        systemIO.FileDelete($"{this.DBFILE}-journal");
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the two directories are equivalent; i.e., they they contain the same subdirectories and files, recursively.
+        /// </summary>
+        private static void AssertDirectoryTreesAreEquivalent(string expectedDir, string actualDir, string contextMessage)
+        {
+            var localMessage = $"{contextMessage}, in directories {expectedDir} and {actualDir}";
+            var expectedSubdirs = systemIO.EnumerateDirectories(expectedDir).OrderBy(systemIO.PathGetFileName);
+            var actualSubdirs = systemIO.EnumerateDirectories(actualDir).OrderBy(systemIO.PathGetFileName);
+            Assert.That(expectedSubdirs.Select(systemIO.PathGetFileName), Is.EquivalentTo(actualSubdirs.Select(systemIO.PathGetFileName)), localMessage);
+            var expectedSubdirsEnumerator = expectedSubdirs.GetEnumerator();
+            var actualSubdirsEnumerator = actualSubdirs.GetEnumerator();
+            while (expectedSubdirsEnumerator.MoveNext() && actualSubdirsEnumerator.MoveNext())
+            {
+                AssertDirectoryTreesAreEquivalent(expectedSubdirsEnumerator.Current, actualSubdirsEnumerator.Current, contextMessage);
+            }
+            var expectedFiles = systemIO.EnumerateFiles(expectedDir).OrderBy(systemIO.PathGetFileName);
+            var actualFiles = systemIO.EnumerateFiles(actualDir).OrderBy(systemIO.PathGetFileName);
+            Assert.That(expectedFiles.Select(systemIO.PathGetFileName), Is.EquivalentTo(actualFiles.Select(systemIO.PathGetFileName)), localMessage);
+            var expectedFilesEnumerator = expectedFiles.GetEnumerator();
+            var actualFilesEnumerator = actualFiles.GetEnumerator();
+            while (expectedFilesEnumerator.MoveNext() && actualFilesEnumerator.MoveNext())
+            {
+                AssertFilesAreEqual(expectedFilesEnumerator.Current, actualFilesEnumerator.Current, contextMessage);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that two files are the same by comparing their size and their contents.
+        /// </summary>
+        private static void AssertFilesAreEqual(string expectedFile, string actualFile, string contextMessage)

```

Comment:
The `TestUtils.CompareFiles` method seems to have a similar intent.  Is there any reason to prefer one over the other?